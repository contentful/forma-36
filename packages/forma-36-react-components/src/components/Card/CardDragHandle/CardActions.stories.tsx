import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import CardDragHandle from './CardDragHandle';

storiesOf('Components|Card/CardDragHandle', module)
  .addParameters({
    propTypes: CardDragHandle['__docgenInfo'],
  })
  .add('default', () => (
    <div style={{ height: 100 }}>
      <CardDragHandle
        className={text('className', '')}
        isDragActive={boolean('isDragActive', false)}
      >
        {text('children', 'CardDragHandle')}
      </CardDragHandle>
    </div>
  ));
