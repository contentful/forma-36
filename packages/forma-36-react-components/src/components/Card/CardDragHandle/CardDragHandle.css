@import 'settings/colors';
@import 'settings/transitions';
@import 'settings/typography';
@import 'settings/borders';

.CardDragHandle {
  display: flex;
  position: relative;
  width: calc(1rem * (20 / var(--font-base-default)));
  height: 100%;
  align-items: center;
  background-color: var(--color-element-lightest);
  border: 0;
  padding: 0;
  border-right: 1px solid var(--color-element-mid);
  cursor: grab;
  border-top-left-radius: var(--border-radius-medium);
  border-bottom-left-radius: var(--border-radius-medium);
  transition: background-color var(--transition-duration-default)
    var(--transition-easing-default);
}

.CardDragHandle:hover,
.CardDragHandle:focus {
  background-color: var(--color-element-light);
}

.CardDragHandle--drag-active {
  background-color: var(--color-element-light);
  cursor: grabbing;
}

.CardDragHandle__sr-label {
  composes: sr-only from './../../../styles/settings/helpers.css';
}
