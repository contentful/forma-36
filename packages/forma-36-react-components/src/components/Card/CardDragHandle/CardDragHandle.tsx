import React from 'react';
import cn from 'classnames';
import { Drag } from '@contentful/f36-icons';

import styles from './CardDragHandle.css';

export interface CardDragHandleProps {
  /**
   * Label rendered as child in CardDragHandle - not visible on screen as
   * purpose is for screen readers only
   */
  children: React.ReactNode;
  /**
   * Class names to be appended to the className prop of the component
   */
  className?: string;
  /**
   * An ID used for testing purposes applied as a data attribute (data-test-id)
   */
  testId?: string;
  /**
   * Applies styling for when the component is actively being dragged by the user
   */
  isDragActive?: boolean;
}

export function CardDragHandle({
  className,
  testId = 'cf-ui-card-drag-handle',
  children,
  isDragActive = false,
  ...otherProps
}: CardDragHandleProps): React.ReactElement {
  const classNames = cn(
    styles.CardDragHandle,
    { [styles['CardDragHandle--drag-active']]: isDragActive },
    className,
  );

  return (
    <div className={classNames} data-test-id={testId} {...otherProps}>
      <Drag variant="muted" />
      <span className={styles['CardDragHandle__sr-label']}>{children}</span>
    </div>
  );
}
