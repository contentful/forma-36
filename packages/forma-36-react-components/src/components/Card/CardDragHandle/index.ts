export * from './CardDragHandle';
export { default } from './CardDragHandle';
