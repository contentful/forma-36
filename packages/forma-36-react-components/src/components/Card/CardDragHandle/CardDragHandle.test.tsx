import React from 'react';
import { shallow, mount } from 'enzyme';
import axe from '../../../utils/axeHelper';
import CardDragHandle from './CardDragHandle';

it('renders the component', () => {
  const output = shallow(<CardDragHandle>CardDragHandle</CardDragHandle>);

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <CardDragHandle className="my-extra-class">CardDragHandle</CardDragHandle>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with isDragActive prop set to true', () => {
  const output = shallow(
    <CardDragHandle isDragActive>CardDragHandle</CardDragHandle>,
  );

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(<CardDragHandle>CardDragHandle</CardDragHandle>).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});
