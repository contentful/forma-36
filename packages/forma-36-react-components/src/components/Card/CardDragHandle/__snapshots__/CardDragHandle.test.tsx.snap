// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  class="CardDragHandle"
  data-test-id="cf-ui-card-drag-handle"
>
  <svg
    class="Icon Icon--small Icon--muted"
    data-test-id="cf-ui-icon"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  <span
    class="CardDragHandle__sr-label"
  >
    CardDragHandle
  </span>
</div>
`;

exports[`renders the component with an additional class name 1`] = `
<div
  class="CardDragHandle my-extra-class"
  data-test-id="cf-ui-card-drag-handle"
>
  <svg
    class="Icon Icon--small Icon--muted"
    data-test-id="cf-ui-icon"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  <span
    class="CardDragHandle__sr-label"
  >
    CardDragHandle
  </span>
</div>
`;

exports[`renders the component with isDragActive prop set to true 1`] = `
<div
  class="CardDragHandle CardDragHandle--drag-active"
  data-test-id="cf-ui-card-drag-handle"
>
  <svg
    class="Icon Icon--small Icon--muted"
    data-test-id="cf-ui-icon"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  <span
    class="CardDragHandle__sr-label"
  >
    CardDragHandle
  </span>
</div>
`;
